### PR TITLE
WIP: Attach the Worker events to the CakePHP Event System

### PR DESCRIPTION
**Pull request by cleptric**
This is WIP for implementing #7 
I started with the Worker events. Queue is missing, but it's only one event.
Also, tests are missing, but I want to discuss this first.

Now you can do sth. like:

```
use Cake\Event\EventManager;

EventManager::instance()->on('Queue.Worker.job.success', function ($event, $job) {
    debug($job->data['job']);
});
```

### CODE REVIEW

File: src/Shell/QueuesadillaShell.php [hunk 1]
```diff
@@ -49,13 +51,17 @@ public function getEngine($logger)
...
+        $workerName = $this->params['worker'];
+        $workerClass = "josegonzalez\\Queuesadilla\\Worker\\" . $workerName . "Worker";
 
-        return new $WorkerClass($engine, $logger, [
+        $worker = new $workerClass($engine, $logger, [
             'queue' => $engine->config('queue'),
             'maxRuntime' => $engine->config('maxRuntime')
         ]);
+
+        $this->__attachEvents($worker);
```
Comment (josegonzalez) [inline src/Shell/QueuesadillaShell.php]: Remove the double-underscore please.

File: src/Shell/QueuesadillaShell.php [hunk 2]
```diff
@@ -92,4 +98,68 @@ public function getOptionParser()
...
+    /**
+     * Attach the league/event events to the CakePHP event system
+     *
+     * @param \josegonzalez\Queuesadilla\Worker\Base $worker worker instance
+     * @return void
+     */
+    private function __attachEvents($worker)
+    {
+        $worker->attachListener('Worker.connectionFailed', function ($event) {
+            $event = new Event('CakeQueuesadilla.Worker.connectionFailed', $this, [
```
Comment (josegonzalez) [inline src/Shell/QueuesadillaShell.php]: Can we change the prefix here to just `Queue`?

File: src/Shell/QueuesadillaShell.php (hunk 1 above)
Comment (cleptric) [inline src/Shell/QueuesadillaShell.php]: Sure, you want to stick with PSR2 I guess?

File: src/Shell/QueuesadillaShell.php (hunk 2 above)
Comment (cleptric) [inline src/Shell/QueuesadillaShell.php]: Sure :)

File: src/Shell/QueuesadillaShell.php (hunk 1 above)
Comment (josegonzalez) [inline src/Shell/QueuesadillaShell.php]: Yep.

**Comment by josegonzalez** [comment]
Needs docs as well :)

File: src/Shell/QueuesadillaShell.php [hunk 3]
```diff
@@ -92,4 +98,68 @@ public function getOptionParser()
...
+
+    /**
+     * Attach the league/event events to the CakePHP event system
+     *
+     * @param \josegonzalez\Queuesadilla\Worker\Base $worker worker instance
+     * @return void
+     */
+    private function attachEvents($worker)
+    {
+        $worker->attachListener('Worker.connectionFailed', function ($event) {
```
Comment (josegonzalez) [inline src/Shell/QueuesadillaShell.php]: These could all be attached in a for loop I think.

**Comment by burzum** [comment]
@cleptric any news on this? :(

**Pull request closed without merging**